packages/react/src/components/Checkbox/__tests__/Checkbox-test.js

```diff
@@ -78,10 +78,20 @@ describe('Checkbox', () => {
     );
   });
 
-  describe('indeterminate', () => {
-    it('should set the indeterminate attribute of the <input> to true', () => {
-      render(<Checkbox id="test" indeterminate labelText="test-label" />);
-      expect(screen.getByRole('checkbox').indeterminate).toBe(true);
-    });
+  it('should NOT call the `onChange` prop when readonly', () => {
```

Comment:
If onClick is disabled as well, this test duplicated but for `onClick` would be nice

Reply:
Added a check for onClick being propagated.